packages/di/src/services/dependency-injection/dependency-scanner.scanner.ts

```diff
@@ -0,0 +1,36 @@
+import 'reflect-metadata';
+
+import { IDependency, IDependencyMap, ProviderClass } from '../../models';
+import { IProviderInject } from '../../models';
+import { CONTAINER_INJECTABLE } from '../../models/constants/containers.constants';
+import { getMetadataDesignParamTypes, getMetadataInject } from '../reflection/reflection';
+
+export class DependencyScanner {
+  #getInjectDecorated(provider: ProviderClass): IProviderInject[] {
+    return [...new Set<IProviderInject>(getMetadataInject(provider))];
+  }
+
+  scan(provider: ProviderClass): IDependencyMap {
+    const dependenciesMap = new Map<ProviderClass, IDependency[]>();
+    const constructorDependency = [...new Set<ProviderClass>(getMetadataDesignParamTypes(provider))];
+
+    const dependencies: IDependency[] = constructorDependency.map((dependency) => {
+      const injectDecoratedDep = this.#getInjectDecorated(provider).find(
+        (injectDependency) => injectDependency.clazz === dependency,
+      );
+      return {
+        provider: dependency,
+        container: injectDecoratedDep ? injectDecoratedDep.store : CONTAINER_INJECTABLE,
+        identifier: injectDecoratedDep ? injectDecoratedDep.identifier : dependency.name,
+      };
```

Comment:
Readability is tiny hard here. To avoid the ternary expression we could use this syntax:
```ts
injectDecoratedDep?.store ?? CONTAINER_INJECTABLE
```

OR

I would even use a regular `if` statement: improves readability, runs null-check once instead of twice, and it makes it easier to add log statements in case of debugging:

```ts
if (injectDecoratedDep) {
  return {
    provider: dependency,
    container: injectDecoratedDep.store,
    identifier: injectDecoratedDep.identifier,
  };
}

return {
  provider: dependency,
  container: CONTAINER_INJECTABLE,
  identifier: dependency.name,
};
```